src/extension.ts

```
@@ -431,6 +440,35 @@ function getJavaConfiguration(): WorkspaceConfiguration {
 	return workspace.getConfiguration('java');
 }
 
+async function cleanWorkspace(workspacePath) {
+	const doIt = 'Restart and delete';
+	window.showWarningMessage('Are you sure you want to clean the Java language server workspace?', 'Cancel', doIt).then(selection => {
+		if (selection === doIt) {
+			const file = path.join(workspacePath, cleanWorkspaceFileName);
+			fs.closeSync(fs.openSync(file, 'w'));
+			commands.executeCommand(Commands.RELOAD_WINDOW);
+		}
+	});
+}
+
+function deleteDirectory(languageClient, dir) {
+	try {
+		if (fs.existsSync(dir)) {
+			fs.readdirSync(dir).forEach(function (child) {
+				let entry = path.join(dir, child);
+				if (fs.lstatSync(entry).isDirectory()) {
+					deleteDirectory(languageClient, entry);
+				} else {
+					fs.unlinkSync(entry);
+				}
+			});
+			fs.rmdirSync(dir);
+		}
+	} catch (error) {
+		languageClient.error(error);
```

actually, since the deletion is recursive, it's prolly better to bubble any exception up (so no try-catch here), and only catch it at the top (L271), with an error displayed to the user. From there I'm not sure if it's safe to proceed with the extension activation.